streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -922,6 +1030,60 @@ private InternalTopicConfig createInternalTopicConfig(final StateStoreSupplier s
     }
 
     /**
+     * Get the Pattern to match all topics requiring to start reading from earliest available offset
+     * @return the Pattern for matching all topics reading from earliest offset, never null
+     */
+    public synchronized Pattern earliestResetTopicsPattern() {
```

Comment:
Currently we avoid overlaps with multiple patterns by assigning the topic to the first pattern that matches, and skip for all other patterns. To be safe, I think it is okay to check that `earliest..` and `latest..` do not have overlaps is OK (they should not, otherwise it is a bug).